src/neuro_api_tony/controller.py

```diff
@@ -192,80 +243,84 @@ def on_unknown_command(self, client_id: int, json_cmd: Any) -> None:
+    def on_view_execute(self, action: NeuroAction) -> bool:
         """Handle an action execution request from the view.
 
         Returns True if an action was sent, False if the action was cancelled.
         """
         if not action.schema:
             # No schema, so send the action immediately
-            self.send_action(
-                client_id,
+            return self.send_action(
+                action.client_id,
                 next(self.id_generator),
                 action.name,
                 None,
             )
-            return True
 
         # If there is a schema, open a dialog to get the data
         result = self.view.show_action_dialog(action)
         if result is None:
             return False  # User cancelled the dialog
 
         self.model.last_action_data[action.name] = result  # Store the last data in the action object
-        self.send_action(client_id, next(self.id_generator), action.name, result)
-        return True
+        return self.send_action(action.client_id, next(self.id_generator), action.name, result)
 
     def on_view_delete_action(self, client_id: int, name: str) -> None:
         """Handle a request to delete an action from the view."""
-        self.model.remove_action_by_name(name)
-        self.view.remove_action_by_name(name)
+        self.model.remove_actions(name=name, client_id=client_id)
+        self.view.remove_actions(name=name, client_id=client_id)
 
         self.view.log_info(f"Action deleted: {name}")
 
-    def on_view_delete_all_actions(self, client_id: int) -> None:
+    def on_view_delete_all_actions(self, client_id: int | None) -> None:
         """Handle a request to delete all actions from the view."""
-        self.model.clear_actions()
-        self.view.clear_actions()
-        self.view.log_info("All actions deleted.")
+        self.model.remove_actions(client_id=client_id)
+        self.view.remove_actions(client_id=client_id)
+        if client_id is not None:
+            game = self.api.get_game_from_client_id(client_id) or f"provisional_name_{client_id}"
+            self.view.log_info(f'All actions deleted for "{game}" (ID: {client_id}).')
+        else:
+            self.view.log_info("All actions deleted.")
 
-    def on_view_unlock(self, client_id: int) -> None:
+    def on_view_unlock(self) -> None:
         """Handle a request to unlock the view."""
         self.view.log_info("Stopped waiting for action result.")  # TODO: Make this configurable as warning?
         self.view.enable_actions()
 
-    def on_view_clear_logs(self, client_id: int) -> None:
+    def on_view_clear_logs(self) -> None:
         """Handle a request to clear the logs from the view."""
         self.view.clear_logs()
         self.view.log_info("Logs cleared.")
 
-    def on_view_send_actions_reregister_all(self, client_id: int) -> None:
+    def on_view_send_actions_reregister_all(self, client_id: int | None) -> None:
```

Comment:
Was expecting it to be `client_id: int | None = None` with a default, curious why not.